.travis.yml

```diff
@@ -11,79 +12,83 @@ matrix:
+      compiler: gcc
+      env: VERSION=6    BUILD=RelWithDebInfo
+    - if: repo = percona/percona-server
+      compiler: gcc
+      env: VERSION=5    BUILD=Debug
+    - if: repo = percona/percona-server
+      compiler: gcc
+      env: VERSION=5    BUILD=RelWithDebInfo
+    - compiler: gcc
+      env: VERSION=4.8  BUILD=Debug
+    - compiler: gcc
+      env: VERSION=4.8  BUILD=RelWithDebInfo
 
 script:
-  - export CC=$GCC
   - JOB_NUMBER=$(echo $TRAVIS_JOB_NUMBER | sed -e 's:[0-9][0-9]*\.\(.*\):\1:')
-  - echo PACKAGES=$PACKAGES PPA=$PPA LLVM=$LLVM JOB_NUMBER=$JOB_NUMBER TRAVIS_BRANCH=$TRAVIS_BRANCH TRAVIS_EVENT_TYPE=$TRAVIS_EVENT_TYPE TRAVIS_PULL_REQUEST=$TRAVIS_PULL_REQUEST
-
-  # Jobs with a number >= 6 are done only for a pull request
-  - if [ $JOB_NUMBER -ge 6 ] && [ "$TRAVIS_EVENT_TYPE" != "pull_request" ]; then 
-       echo Finishing as this is not a pull request; 
-       travis_terminate 0; 
-    fi
+  - echo PACKAGES=$PACKAGES JOB_NUMBER=$JOB_NUMBER TRAVIS_REPO_SLUG=$TRAVIS_REPO_SLUG TRAVIS_BRANCH=$TRAVIS_BRANCH TRAVIS_EVENT_TYPE=$TRAVIS_EVENT_TYPE TRAVIS_PULL_REQUEST=$TRAVIS_PULL_REQUEST
 
   # Update required LLVM and Ubuntu Toolchain repositories
-  - if [ "$LLVM" != "" ]; then
+  - if [ "$CC" == "clang" ]; then
+       PACKAGES="$PACKAGES llvm-$VERSION-dev";
        curl -sSL "http://apt.llvm.org/llvm-snapshot.gpg.key" | sudo -E apt-key add -;
-       echo "deb http://apt.llvm.org/trusty/ $LLVM main" | sudo tee -a /etc/apt/sources.list > /dev/null;
+       echo "deb http://apt.llvm.org/trusty/ llvm-toolchain-trusty-$VERSION main" | sudo tee -a /etc/apt/sources.list > /dev/null;
+    else
+       sudo -E apt-add-repository -y "ppa:ubuntu-toolchain-r/test";
     fi
-  - if [ "$PPA" != "" ]; then
-       sudo -E apt-add-repository -y "ppa:$PPA";
-    fi
-  - if [ "$LLVM" != "" ] || [ "$PPA" != "" ]; then
-       sudo -E apt-get -yq update &>> ~/apt-get-update.log;
-    fi 
+  - sudo -E apt-get -yq update &>> ~/apt-get-update.log
 
   # Download dependencies
-  - PACKAGES="$PACKAGES cmake cmake-curses-gui libaio-dev libssl-dev libncurses5-dev libevent-dev bison"
-  - sudo -E apt-get -yq --no-install-suggests --no-install-recommends install $PACKAGES
+  - CC=$CC-$VERSION
+  - CXX=$CXX-$VERSION
+  - sudo -E apt-get -yq --no-install-suggests --no-install-recommends install $CXX $PACKAGES cmake cmake-curses-gui libaio-dev libssl-dev libncurses5-dev libevent-dev liblz4-dev bison
   - mkdir bin; cd bin
   - $CC -v
   - $CXX -v
 
   # Test "RelWithDebInfo" compilation
   - cmake ..
-    -DCMAKE_BUILD_TYPE=RelWithDebInfo
+    -DCMAKE_BUILD_TYPE=$BUILD
     -DMYSQL_MAINTAINER_MODE=ON
     -DBUILD_CONFIG=mysql_release
     -DFEATURE_SET=community
     -DENABLE_DTRACE=OFF
-    -DWITH_SSL=$LIBTYPE
-    -DWITH_ZLIB=$LIBTYPE
-    -DWITH_LIBEVENT=$LIBTYPE
     -DENABLE_DOWNLOADS=1
     -DDOWNLOAD_BOOST=1
     -DWITH_BOOST=../deps
     -DWITHOUT_TOKUDB=ON
     -DWITHOUT_ROCKSDB=ON
+    -DWITH_RAPID=OFF
      $CMAKE_OPT
```

Comment:
Depending on the status of #2123 it also make sense to add `-DWITH_KEYRING_VAULT=ON` here.